### PR TITLE
Add routes that allow the user to exchange basic auth credentials for a session

### LINKED ISSUE #890: Add user session handler

### PR DESCRIPTION
**Pull request by desa**
Closes #890

_Briefly describe your proposed changes:_
This PR adds a session service that is intended to be used after a user has authenticated. A session should represent a users maximal privileges, although this has yet to be fully worked out. It contains an ID so that it will be auditable.

This PR adds `/signin` and `/signout` routes that allow the user to exchange basic auth credentials for a session.

  - [ ] Rebased/mergeable
  - [ ] Tests pass
  - [ ] swagger.json updated (if modified Go structs or API)

### CODE REVIEW

**Comment by kelwang** [comment]
I don't think store each session entry with expire time in boltdb is a good idea. Usually session should only end if no new activity. It also means we have to keep lock and unlock the boltdb for every users' activities.  I suggest we could have a sessionManager with 2 inmem sync.Map. One for time(each minute as key)->users, another one for user->time. The session Manager will try to expire the session every minute, same time user can update their own user->time index, then update the main index.  In a separated go routine, the boltdb will try to store the whole 2 indexes every minute, So when the application restart, user don't have to re-login.

**Comment by desa** [comment]
@kelwang 

> Usually session should only end if no new activity. 

This depends on product requirements. There are plenty of services that maintain long lived sessions so that you don't have the re-authenticate. My thinking was that for the alpha it would be better for the user to not have to re-authenticate frequently. What do you think?

> It also means we have to keep lock and unlock the boltdb for every users' activities. I suggest we could have a sessionManager with 2 inmem sync.Map. One for time(each minute as key)->users, another one for user->time. The session Manager will try to expire the session every minute, same time user can update their own user->time index, then update the main index. 

I'm not too concerned with performance at the moment (authorizations have the same lookup path currently in platform), but when it become a concern we'll definitely need to do something like this. Maybe there's some way we can tweek the `SessionService` interface that will lend itself to this type of implementation?

**Comment by kelwang** [comment]
Yeah, I agree it should be cool for now

File: http/session_handler.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	BasicAuthService platform.BasicAuthService
+	SessionService   platform.SessionService
+}
+
+// NewSessionHandler returns a new instance of SessionHandler.
+func NewSessionHandler() *SessionHandler {
+	h := &SessionHandler{
+		Router: httprouter.New(),
+	}
+
```
Comment (jademcgough) [inline http/session_handler.go]: Should we document these endpoints in the swagger doc?

File: http/session_handler.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+		EncodeError(ctx, err, w)
+		return
+	}
+
+	if err := h.SessionService.ExpireSession(ctx, req.Key); err != nil {
+		EncodeError(ctx, err, w)
+		return
+	}
+
+	// TODO(desa): not sure what to do here maybe redirect?
```
Comment (jademcgough) [inline http/session_handler.go]: IMO this is fine for now

File: http/session_handler.go (hunk 1 above)
Comment (desa) [reply http/session_handler.go]: yup

**Pull request merged**